python_modules/libraries/dagster-k8s/dagster_k8s/external_resource.py

```diff
@@ -0,0 +1,189 @@
+import json
+import random
+import string
+from contextlib import contextmanager
+from typing import Iterator, Mapping, Optional, Sequence, Union
+
+import kubernetes
+from dagster import OpExecutionContext
+from dagster._core.external_execution.context import (
+    ExternalExecutionOrchestrationContext,
+)
+from dagster._core.external_execution.resource import (
+    ExternalExecutionContextInjector,
+    ExternalExecutionMessageReader,
+    ExternalExecutionParams,
+    ExternalExecutionResource,
+)
+from dagster._core.external_execution.utils import (
+    ExternalExecutionEnvContextInjector,
+    io_params_as_env_vars,
+)
+from dagster_externals import (
+    ExternalExecutionExtras,
+)
+
+from .client import DagsterKubernetesClient, WaitForPodState
+
+
+def get_pod_name(run_id: str, op_name: str):
+    clean_op_name = op_name.replace("_", "-")
+    suffix = "".join(random.choice(string.digits) for i in range(10))
+    return f"dagster-{run_id[:18]}-{clean_op_name[:20]}-{suffix}"
+
+
+DEFAULT_TIMEOUT = 30
+
+
+class K8sPodLogsMessageReader(ExternalExecutionMessageReader):
+    @contextmanager
+    def read_messages(
+        self,
+        _context: ExternalExecutionOrchestrationContext,
+    ) -> Iterator[ExternalExecutionParams]:
+        yield {}
+
+    def consume_pod_logs(
+        self,
+        external_context: ExternalExecutionOrchestrationContext,
+        client: DagsterKubernetesClient,
+        pod_name: str,
+        namespace: str,
+    ):
+        for line in client.core_api.read_namespaced_pod_log(
+            pod_name,
+            namespace,
+            follow=True,
+            _preload_content=False,  # avoid JSON processing
+        ).stream():
+            log_chunk = line.decode("utf-8")
+            for log_line in log_chunk.split("\n"):
+                # exceptions as control flow, you love to see it
+                try:
+                    message = json.loads(log_line)
+                    # need better message check
+                    assert message.keys() == {"method", "params"}
+                    external_context.handle_message(message)
+                except Exception:
+                    # move non-message logs in to stdout for compute log capture
+                    print(log_line)  # noqa: T201
```

Comment:
this is out of line with the other impls so far - since it depends directly on the output of the invocation process